### PR TITLE
fix given

### PR DESCRIPTION
**Pull request by sanketXynes**
1. Indeterminate state added
2. isChecked removed, onUpdate fixed.
3. Behavior of checkbox fixed

### CODE REVIEW

File: src/components/Checkbox/checkbox.stories.ts [hunk 1]
```diff
@@ -18,6 +18,9 @@ export const CheckBox: Story = {
   args: {
     label: "breakfast",
     value: false,
+    onUpdate: (e) => {
+      console.log("The val------", e);
```
Comment (archan96) [inline src/components/Checkbox/checkbox.stories.ts]: Remove logs

**Review by archan96 — CHANGES_REQUESTED** [review]
@sanketXynes please make the changes

**Review by Aman9804 — APPROVED** [review]
approved

**Review by archan96 — APPROVED** [review]
Looks good.

**Review by xynes-admin — APPROVED** [review]
Looks good

**Pull request merged**